### ISSUE TITLE
flash-attention onnx export.

### ISSUE THREAD

**Issue by scuizhibin** [onnx/onnx#6359]
When using Flash-Attention version 2.6.3, there is an issue with the ONNX file saved using torch.onnx.export.
code:
 import sys
 import torch 
qkv=torch.load("/home/qkv.pth") 
from  modeling_intern_vit import FlashAttention
falsh=FlashAttention().eval().cuda()
out=falsh(qkv.cpu().cuda())

with torch.no_grad():
   torch.onnx.export(
            falsh, 
            (qkv,),
              "/home/qkv.onnx",
             input_names   = ["input0"],
            output_names  = ["qkv_out"],
            opset_version = 11        )
![image](https://github.com/user-attachments/assets/9c489934-7165-43d1-88f4-8df8eb757060)

**Comment by xadupre** [comment]
Do you see any error message? Can you export without FlashAttention?

**Comment by justinchuby** [comment]
Please provide the full definition of FlashAttention and a script that can run directly to reproduce the issue. Refer to https://www.markdownguide.org/basic-syntax/ to format the code block. If there are any accuracy issues, please also provide the full code snippet that shows those issues.
